components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1159,7 +1063,9 @@ void AdsServiceImpl::ShutdownAdsService() {
   bat_ads_client_associated_receiver_.reset();
   bat_ads_service_remote_.reset();
 
-  url_loaders_.clear();
+  if (network_client_) {
```

Comment:
can `network_client_` be null?

Reply:
In followup PRs yes if tests do not care about network requests then yes this will be null.

Reply:
Added, thanks